Source/WebKit/NetworkProcess/curl/NetworkDataTaskCurl.cpp

```diff
@@ -283,6 +289,36 @@ bool NetworkDataTaskCurl::shouldRedirectAsGET(const ResourceRequest& request, bo
     return false;
 }
 
+void NetworkDataTaskCurl::didReadDataURL(std::optional<DataURLDecoder::Result>&& result)
```

Comment:
It seems that there are cases where result is not returned.

https://github.com/WebKit/WebKit/blob/d488f398f95d05b533bb5a45084b49d91c36ee84/Source/WebKit/NetworkProcess/soup/NetworkDataTaskSoup.cpp#L1802-L1805

Reply:
Ah, you are checking with downloadDataURL. However, since it is useless to execute didReceiveResponse, is it possible to execute failure processing inside this function?
NetworkDataTaskSoup::didFailDownload may be helpful.

Reply:
Yes, I can certainly move it.